### PR TITLE
upgrade qemu to 2.1.3

### PR DESCRIPTION
**Pull request by ddcc**

### CODE REVIEW

**Comment by nedwill** [comment]
Did you check that the patches in qemu_mods/ work for 2.1.3? If not, I can take a look later and port them.

**Comment by ddcc** [comment]
The patches apply fine and QIRA seems to work (at least for /bin/ls), but I'm not super familiar with QIRA to know if it's 100% correct.

**Comment by geohot** [comment]
So I apply the QEMU patches in a really hacky way, I just copy over the files(see tracers/qemu_mods). So there's files from 2.1.0rc being copied into 2.1.3

To do this right, change the QEMU patching to use patch(and leave a script in extra to regenerate the patchfile), generate the patch on the current version, and apply that to the new one.

**Comment by ddcc** [comment]
Sorry, I was under the impression that it was using patch, guess I should have taken a closer look. This should be fixed now.

File: tracers/qemu_build.sh [hunk 1]
```diff
@@ -18,34 +20,16 @@ if [ ! -d qemu/qemu-latest ]; then
   rm -rf qemu
   mkdir -p qemu
   cd qemu
-  wget http://wiki.qemu-project.org/download/qemu-2.1.0-rc0.tar.bz2
-  tar xf qemu-2.1.0-rc0.tar.bz2
-  ln -s qemu-2.1.0-rc0 qemu-latest
-
-  ln -s qemu-latest/arm-linux-user/qemu-arm qira-arm
```
Comment (geohot) [inline tracers/qemu_build.sh]: these symlinks can't be removed

**Comment by geohot** [comment]
The qemu-latest/arm-linux-user/qemu-... qira-... symlinks are still needed, otherwise looks good. I'm shocked the tests passed without them, that's a bug in the tests.

**Comment by ddcc** [comment]
Fixed.

**Comment by geohot** [comment]
LGTM

If you are feeling ambitious, https://travis-ci.org/BinaryAnalysisPlatform/qira/builds/55660964 should have failed without QEMU found, not sure why it didn't.

**Comment by ddcc** [comment]
I don't have time to look at it, but I believe you have two problems:
1. `run_tests.sh` runs QIRA in the background using the ampersand operator, so even if it returns a non-zero error code, it won't be caught by `bash -e`. For example, compare the output of `false; echo $?` and `false & echo $?`
2. `qira.py` has some sort of exception handler (or it's multithreaded), because it's still starting the webserver despite the exception in `program.execqira`

**Pull request merged**